### PR TITLE
feat: analytics service provide

### PR DESCRIPTION
**Pull request by AjimenezDCL**
#299 

Goes in pair with this `explorer` PR: https://github.com/decentraland/explorer/pull/2369

It cannot be tested since no implementation is done yet. A smoke test should be enough

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/WebInterface/Interface.cs [hunk 1]
```diff
@@ -1186,5 +1194,7 @@ public static void RequestUserProfile(string userId)
         }
 
         public static void ReportAvatarFatalError() { SendMessage("ReportAvatarFatalError"); }
+
+        public static void SendGenericAnalytic(string eventName, Dictionary<object, object> data) { SendMessage("GenericAnalytic", new GenericAnalyticPayload { eventName = eventName, data = data }); }
```
Comment (pbosio) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/WebInterface/Interface.cs:1198]: There is already a `ReportAnalyticsEvent` method in this class.
Is replacing/removing that planned for a following PR?

**Review by pbosio — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/ServiceProviders/Analytics/IAnalytics.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+public class Analytics : IAnalytics
+{
+    //Remove this once environment is on its own assembly and can be accessed properly
+    public static IAnalytics i;
+    public Analytics() { i = this; }
+
+    public void SendAnalytic(string eventName, Dictionary<object, object> data) { SendToSegment(eventName, data); }
+
+    internal void SendToSegment(string eventName, Dictionary<object, object> data) { WebInterface.SendGenericAnalytic(eventName, data); }
```
Comment (pbosio) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/ServiceProviders/Analytics/IAnalytics.cs:18]: Could we add a `VERBOSE` flag to be able to look at console's log what analytics are being send?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/WebInterface/Interface.cs (hunk 1 above)
Comment (AjimenezDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/WebInterface/Interface.cs:1198]: I didnt know about that method. Gonna dig a bit further, maybe we can reuse this and the `kernel` PR is not even needed.

Comment (AjimenezDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/WebInterface/Interface.cs]: You were right @pbosio , the existent endpoint works as well here. I'm gonna use it!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/ServiceProviders/Analytics/IAnalytics.cs (hunk 2 above)
Comment (AjimenezDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/ServiceProviders/Analytics/IAnalytics.cs]: That's a great idea, thanks!

**Review by pravusjif — APPROVED** [review]

**Pull request merged**